src/disasm/section.rs

```diff
@@ -0,0 +1,257 @@
+/*
+ * This file is part of Disasm-Util.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Access to the Section struct.
+//!
+//! This module contains the Section struct which is a named collection of symbols.
+use std::fmt;
+
+use super::Instruction;
+use super::Symbol;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Section {
+    name: String,
+    symbols: Vec<Symbol>,
+}
+
+impl Section {
+    pub fn new(name: &str) -> Self {
+        Section {
+            name: name.trim().to_string(),
+            symbols: Vec::new(),
+        }
+    }
+
+    pub fn add_symbol(&mut self, symbol: Symbol) {
+        self.symbols.push(symbol);
+    }
+
+    pub fn add_instruction(&mut self, instruction: Instruction) -> Result<(), String> {
+        self.symbols
+            .last_mut()
+            .ok_or("Attempted to add an instruction without first defining a symbol")?
+            .add_instruction(instruction);
+        Ok(())
+    }
+
+    pub fn get_name(&self) -> &String {
+        &self.name
+    }
+
+    pub fn sort_symbols(&mut self) {
+        self.symbols.sort_by(|a, b| a.get_name().cmp(b.get_name()));
+    }
+}
+
+impl fmt::Display for Section {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let joined = self
+            .symbols
+            .iter()
+            .map(|sym| {
+                sym.to_string()
+                    .split('\n')
+                    .map(|s| {
+                        if s.is_empty() {
+                            format!("")
+                        } else {
+                            format!("    {}\n", s)
+                        }
+                    })
+                    .collect::<Vec<_>>()
+                    .join("")
+            })
```

Comment:
Try to factor out some of this code in a new function to improve readability

Reply:
I cleaned it up a little, let me know if you think it looks better now.